### ISSUE TITLE
$genericblock exceptions don't work until you restart Safari

### ISSUE THREAD

**Issue by vozersky** [AdguardTeam/AdguardBrowserExtension#508]
STR:

1. Add:
`@@||4players.de^$genericblock`

2. Open http://www.4players.de/
3. Check console.

![image](https://cloud.githubusercontent.com/assets/17472907/21813728/370bb706-d768-11e6-97db-1eec2fe11413.png)

4. Restart Safari and check again, now the exception works.

**Comment by vozersky** [comment]
@Mizzick can it be fixed?

**Comment by atropnikov** [comment]
After running the extension
```
[Info] 2017-01-11T09:36:52.168Z: Starting loading content blocker. (log.js, line 71)
[Info] 2017-01-11T09:36:52.252Z: Safari Content Blocker Converter v1.3.20 (log.js, line 71)
[Info] 2017-01-11T09:36:52.252Z: Converting 35101 rules. Optimize=false (log.js, line 71)
[Info] 2017-01-11T09:36:52.538Z: Applying 1036 css exceptions (log.js, line 71)
[Info] 2017-01-11T09:36:52.557Z: Css exceptions applied: 818 (log.js, line 71)
[Info] 2017-01-11T09:36:52.557Z: Css exceptions errors: 18 (log.js, line 71)
[Info] 2017-01-11T09:36:52.557Z: Trying to compact 13753 elemhide rules (log.js, line 71)
[Info] 2017-01-11T09:36:52.559Z: Compacted result: wide=22 domainSensitive=7671 (log.js, line 71)
[Info] 2017-01-11T09:36:52.559Z: Rules converted: 33705 (1396 errors) (log.js, line 71)
Basic rules: 13867
Elemhide rules (wide): 22
Elemhide rules (generic domain sensitive): 710
Exceptions Elemhide (wide): 573
Elemhide rules (domain-sensitive): 7671
Exceptions (elemhide): 193
Exceptions (other): 4265
[Info] 2017-01-11T09:36:52.570Z: Content blocker length: 27301 (log.js, line 71)
[Info] 2017-01-11T09:36:52.761Z: Setting content blocker. Length=27301 (log.js, line 71)
[Info] 2017-01-11T09:36:52.896Z: Content blocker has been set. (log.js, line 71)
```
After adding rule
```
@@||4players.de^$genericblock
```
```
[Info] 2017-01-11T09:37:18.563Z: Starting loading content blocker. (log.js, line 71)
[Info] 2017-01-11T09:37:18.652Z: Safari Content Blocker Converter v1.3.20 (log.js, line 71)
[Info] 2017-01-11T09:37:18.652Z: Converting 35102 rules. Optimize=false (log.js, line 71)
[Info] 2017-01-11T09:37:18.868Z: Applying 1036 css exceptions (log.js, line 71)
[Info] 2017-01-11T09:37:18.885Z: Css exceptions applied: 818 (log.js, line 71)
[Info] 2017-01-11T09:37:18.885Z: Css exceptions errors: 18 (log.js, line 71)
[Info] 2017-01-11T09:37:18.886Z: Trying to compact 13753 elemhide rules (log.js, line 71)
[Info] 2017-01-11T09:37:18.888Z: Compacted result: wide=22 domainSensitive=7671 (log.js, line 71)
[Info] 2017-01-11T09:37:18.888Z: Rules converted: 33705 (1397 errors) (log.js, line 71)
Basic rules: 13867
Elemhide rules (wide): 22
Elemhide rules (generic domain sensitive): 710
Exceptions Elemhide (wide): 572
Elemhide rules (domain-sensitive): 7671
Exceptions (elemhide): 193
Exceptions (other): 4266
[Info] 2017-01-11T09:37:18.902Z: Content blocker length: 27301 (log.js, line 71)
[Info] 2017-01-11T09:37:19.089Z: Setting content blocker. Length=27301 (log.js, line 71)
[Info] 2017-01-11T09:37:19.270Z: Content blocker has been set. (log.js, line 71)
```

**Comment by Mizzick** [comment]
Fixed with #510

**Comment by vozersky** [comment]
Looks like it works fine now (2.5.11 Dev).
